buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -1242,5 +1242,16 @@ private static boolean isUtf8(ByteBuf buf, int index, int length) {
         return true;
     }
 
+    public static int lengthOfByteOrderMark(ByteBuf buf, Charset charset) {
+        final byte[] bom = "\uFEFF".getBytes(charset);
```

Comment:
This is surprisingly expensive to create, at least two allocs if memory serves.  You should make this a final static var for UTF-8 and ASCII

Reply:
Done